whelktool/scripts/cleanups/2020/05/lxl-3139-text-to-BF2.groovy

```diff
@@ -0,0 +1,58 @@
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+
+String where = "data#>>'{@graph,1,@type}' = 'TextInstance' and data#>>'{@graph,2,@type}' = 'Text'"
+
+selectBySqlWhere(where) { data ->
+    def (record, instance, work) = data.graph
+    boolean changed = false
+
+    if (instance["marc:mediaTerm"] == null) {
+        List carrierTypes = instance["carrierType"]
+        for (Object carrierType : carrierTypes) {
+            if (carrierType["@id"] == "https://id.kb.se/marc/RegularPrint" ||
+                    carrierType["@id"] == "https://id.kb.se/marc/RegularPrintReproduction" ||
+                    carrierType["@id"] == "https://id.kb.se/term/rda/Volume" ||
+                    carrierType["@id"] == "https://id.kb.se/marc/LargePrint" ||
+                    carrierType["@id"] == "https://id.kb.se/marc/RegularPrintReproductionEyeReadablePrint") {
+                instance["@type"] = "Print"
+                changed = true
+            }
+        }
+    }
+
+    else if (instance["marc:mediaTerm"].contains("Elektronisk") || instance["marc:mediaTerm"].contains("Electronic")) {
+        instance["@type"] = "Electronic"
+        changed = true
+    }
+
+    else if (instance["marc:mediaTerm"].contains("handskrift") || instance["marc:mediaTerm"].contains("Handskrift")) {
+        instance["@type"] = "Manuscript"
+        changed = true
+    }
+
+    else if (instance["marc:mediaTerm"].contains("Kombinerat") || instance["marc:mediaTerm"].contains("kombinerat")) {
+        instance["@type"] = "Instance"
+        changed = true
+    }
+
+    else if (instance["hasPart"]) {
```

Comment:
This seems to assume that something with a printed text part makes the whole thing a printed text?

Reply:
Thus the "careful review point" :) The spec says:

in '{@graph,1,hasPart,@type}' =  'TextInstance'
     'Print' if 'carrierType' = https://id.kb.se/marc/RegularPrint
I guess its the "part" that should have type Print then?